<a href="https://colab.research.google.com/github/lajmej/Analisis_Numerico/blob/main/Tarea4/Tarea4Ejercicio8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Ejercicio 8

**Integrantes:**



*   De la Luz Siles Cristobal
*   Mejia Quinarez Luis Eduardo


**Mostrar las primeras cinco iteraciones del Método de Newton para hallar el cero de $f(x)=x^{2}-6$ con $x_{0}=1$.**

In [ ]:
def Newton(f,df,Tol,N,x0):
  Iter=0

  while Iter<=N:
    fx=f(x0)
    dfx=df(x0)

    xn = x0 - (fx/float(dfx))

    print("Iteración número " + str(Iter+1) + ", resultado= "  + str(xn))
    if abs(f(xn))<=Tol and abs(xn-x0)<=Tol:
      print("Tu raiz es "+str(xn))
      return xn

    x0=xn

    Iter+=1

  else:
    print("El valor aproximado de tu raiz es "+str(xn))

In [ ]:
f=lambda x: (x**2) -6

df=lambda x: 2*x

FB=Newton(f,df,0.000001,4,1)

Iteración número 1, resultado= 3.5
Iteración número 2, resultado= 2.607142857142857
Iteración número 3, resultado= 2.454256360078278
Iteración número 4, resultado= 2.4494943716069653
Iteración número 5, resultado= 2.4494897427875517
El valor aproximado de tu raiz es 2.4494897427875517


*   **Tolerancia=** *$\frac{1}{100000}$*
*   **Número de iteraciones=** *$100$*
*   **$f(x)=x^{2} -6$**
*   **$f^{'}(x)=2x$**


*Primera iteración: iteracion=1*

**$x_{0}$=1**

---


**Mientras $iteracion=1 \leq 100$:**

**$f(x_{0})=-5$**

**$f^{'}(x_{0})=2$**

**$x_{1}=x_{0}-\frac{f(x_{0})}{f^{'}(x_{0})}=3.5$**

**Si $|f(x_{1})|=6.25\leq \frac{1}{1000}$ y $|x_{1}-x_{0}|=2.5\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.5,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$iteracion=iterancion+1=2$**

*Segunda iteración: iteracion=2*

**$x_{0}$=3.5**

---


**Mientras $iteracion=2 \leq 100$:**

**$f(x_{0})=6.26$**

**$f^{'}(x_{0})=7.0$**

**$x_{1}=x_{0}-\frac{f(x_{0})}{f^{'}(x_{0})}=3.5$**

**Si $|f(x_{1})|=0.79719\leq \frac{1}{1000}$ y $|x_{1}-x_{0}|=0.8928\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=3.5,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$iteracion=iterancion+1=3$**

*Tercera iteración: iteracion=3*

**$x_{0}$=2.6071**

---


**Mientras $iteracion=3 \leq 100$:**

**$f(x_{0})=0.7971$**

**$f^{'}(x_{0})=5.21428$**

**$x_{1}=x_{0}-\frac{f(x_{0})}{f^{'}(x_{0})}=2.4542$**

**Si $|f(x_{1})|=0.023374\leq \frac{1}{1000}$ y $|x_{1}-x_{0}|=0.1528\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=2.4542,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$iteracion=iterancion+1=4$**

*Cuarta iteración: iteracion=4*

**$x_{0}$=2.4542**

---


**Mientras $iteracion=4 \leq 100$:**

**$f(x_{0})=0.0233$**

**$f^{'}(x_{0})=4.9085$**

**$x_{1}=x_{0}-\frac{f(x_{0})}{f^{'}(x_{0})}=2.4494$**

**Si $|f(x_{1})|=2.2676\leq \frac{1}{1000}$ y $|x_{1}-x_{0}|=0.0047\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=2.4494,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$iteracion=iterancion+1=5$**

*Quinta iteración: iteracion=5*

**$x_{0}$=2.4494943716069653**

---


**Mientras $iteracion=5 \leq 100$:**

**$f(x_{0})=2.2676534201515608e-05$**

**$f^{'}(x_{0})=4.8989887432139305$**

**$x_{1}=x_{0}-\frac{f(x_{0})}{f^{'}(x_{0})}=2.4494897427875517$**

**Si $|f(x_{1})|=2.142641619684582e-11\leq \frac{1}{1000}$ y $|x_{1}-x_{0}|=4.628819413543539e-06\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=2.4494,$**
**Se cumple.**

**Se regresa $x_{1}$**
